packages/checkout/widgets-lib/src/lib/hooks/useInjectedProviders.ts

```diff
@@ -0,0 +1,87 @@
+import { useCallback, useEffect, useState } from 'react';
+import { EIP6963ProviderDetail } from 'mipd/src/types';
+import { Checkout, WalletProviderRdns } from '@imtbl/checkout-sdk';
+import { EIP1193Provider } from 'mipd';
+import { Web3Provider } from '@ethersproject/providers';
+import { getPassportProviderDetail, InjectedProvidersManager } from '../eip6963';
+
+export interface UseInjectedProvidersParams {
+  checkout: Checkout | null;
+}
+
+let passportWeb3Provider: Web3Provider;
+const processProviders = (checkout: Checkout | null, injectedProviders: EIP6963ProviderDetail<EIP1193Provider>[]) => {
+  // Apply wallet providers allowlist
+  const allowedWalletRdns = [
+    'com.immutable.passport', // Immutable Passport
+    'io.metamask', // MetaMask
+    // 'xyz.frontier.wallet', // Frontier
+    // 'me.rainbow', // Rainbow
+    // 'com.coinbase.wallet', // Coinbase Wallet
+  ];
+  const uniqueRdnsSet = new Set();
+  const filteredProviders = injectedProviders
+    .filter((provider) => {
+      if (allowedWalletRdns.includes(provider.info.rdns)) {
+        if (!uniqueRdnsSet.has(provider.info.rdns)) {
+          uniqueRdnsSet.add(provider.info.rdns);
+          return true;
+        }
+      }
+
+      return false;
+    })
+    .sort((a, b) => {
+      // Get the index of the rdns for each provider
+      const indexA = allowedWalletRdns.indexOf(a.info.rdns);
+      const indexB = allowedWalletRdns.indexOf(b.info.rdns);
+
+      // Sort based on the index
+      return indexA - indexB;
+    });
+
+  // Add passport from checkout config if not from injected providers
+  if (checkout?.passport
+    && allowedWalletRdns.includes(WalletProviderRdns.PASSPORT)
+    && !filteredProviders.some((provider) => provider.info.rdns === WalletProviderRdns.PASSPORT)) {
+    if (!passportWeb3Provider) {
+      passportWeb3Provider = new Web3Provider(checkout.passport.connectEvm());
+    }
+    filteredProviders.unshift(getPassportProviderDetail(passportWeb3Provider.provider as EIP1193Provider));
+  }
+
+  return filteredProviders;
+};
+
+/**
+ * Hook that supplies a filters and sorted list of EIP-6963 injected providers.
+ */
+export const useInjectedProviders = ({ checkout }: UseInjectedProvidersParams) => {
```

Comment:
So just making sure I understand what we're doing here, every time we use this hook we get an instance of the InjectedProvidersManager, that fires the event to call for all of the providers to reveal themselves. Then we add the Passport provider into that list if it hasn't been revealed by the 6963 event. Then we store the state for the passport provider at the top of this file and only create it once.

Reply:
Yep exactly.. the idea is that we support a fallback mechanism and then I'll work with passport SDK to add support for EIP6963 and the instance will be managed elsewhere then.